lib/db/mysql.js

```diff
@@ -1451,33 +1456,36 @@ module.exports = function (log, error) {
   }
 
   const DELETE_RECOVERY_CODES = 'CALL deleteRecoveryCodes_1(?)'
-  const INSERT_RECOVERY_CODE = 'CALL createRecoveryCode_1(?, ?)'
+  const INSERT_RECOVERY_CODE = 'CALL createRecoveryCode_2(?, ?, ?)'
   MySql.prototype.replaceRecoveryCodes = function (uid, count) {
 
     // Because of the hashing requirements the process of replacing
     // recovery codes is done is two separate procedures. First one
     // deletes all current codes and the second one inserts the
     // hashed randomly generated codes.
-    return dbUtil.generateRecoveryCodes(count)
-      .then((codeList) => {
+    return dbUtil.generateRecoveryCodes(count, RECOVERY_CODE_KEYSPACE, RECOVERY_CODE_LENGTH)
+      .then((codes) => {
         return this.read(DELETE_RECOVERY_CODES, [uid])
           .then(() => {
-            if (codeList <= 0) {
+            if (codes <= 0) {
               return P.resolve([])
             }
 
-            const queries = []
-            codeList.forEach((code) => {
-              queries.push({
-                sql: INSERT_RECOVERY_CODE,
-                params: [uid, dbUtil.createHashSha512(code)]
-              })
-            })
+            return Promise.all(codes.map((code) => dbUtil.createHashScrypt(code)))
+              .then((items) => {
+                const queries = []
+                items.forEach((item) => {
+                  queries.push({
+                    sql: INSERT_RECOVERY_CODE,
+                    params: [uid, item.hash, item.salt]
+                  })
+                })
 
-            return this.writeMultiple(queries)
+                return this.writeMultiple(queries)
+              })
           })
           .then(() => {
-            return P.resolve(codeList)
+            return P.resolve(codes)
```

Comment:
Again, realise it was here before but the `P.resolve` is unnecessary here, could just be `.then(() => codes)`.